sympy/controlsys/lti.py

```diff
@@ -0,0 +1,1363 @@
+        sympy.matrices.matrices.ShapeError: Dimensions of inputs and outputs must match!
+
+        See Also
+        ========
+
+        cascade: cascade interconnection of two systems
+        """
+        if not isinstance(other, StateSpaceModel):
+            raise TypeError("Argument must be of type StateSpaceModel, not %r" % (type(other)))
+        # assert matching shapes
+        if not ((self.represent[1].shape[1] == other.represent[1].shape[1]) and
+                (self.represent[2].shape[0] == other.represent[2].shape[0])):
+            raise ShapeError("Dimensions of inputs and outputs must match!")
+
+        newA = self.represent[0].row_join(zeros(self.represent[0].rows, other.represent[0].cols)) \
+                                .col_join(
+                                    zeros(other.represent[0].rows, self.represent[0].cols)
+                                    .row_join(other.represent[0]))
+        newB = self.represent[1].col_join(other.represent[1])
+        newC = self.represent[2].row_join(other.represent[2])
+        newD = self.represent[3] + other.represent[3]
+
+        return StateSpaceModel(newA, newB, newC, newD)
+
+    def __eq__(self, other):
+        if isinstance(other, StateSpaceModel):
+            return TransferFunctionModel(self) == TransferFunctionModel(other)
+        elif isinstance(other, TransferFunctionModel):
+            return TransferFunctionModel(self) == other
+        return NotImplemented
+
+    #
+    # define a magic function for unknown method handling
+    #   the class tries to pass the method to the matrices in self.represent
+    def __getattr__(self, name):
+
+        # dont overwrite private or magic function attribute testing!
+        if name[0] == '_':
+            raise AttributeError("%r object has no attribute %r" %
+                                 (self.__class__, name))
+
+        try:
+            def handler(*args, **kwargs):
+
+                new_represent = []
+                for r in self.represent:
+                    methodToCall = getattr(r, name)
+                    new_represent.append(methodToCall(*args, **kwargs))
+                return StateSpaceModel(*new_represent)
+
+        except AttributeError:
+            raise AttributeError("%r object has no attribute %r" %
+                                 (self.__class__, name))
+
+        return handler
+
+    #
+    # _repr_latex_(self)
+    #   defines the representation of the class in ipython pretty printing
+    #
+    def _repr_latex_(self):
+        return '$' + latex(self._blockrepresent) + '$'
+
+    def __str__(self):
+        return 'StateSpaceModel(\n' + sstr(self.represent[0]) + ',\n' \
+                                    + sstr(self.represent[1]) + ',\n' \
+                                    + sstr(self.represent[2]) + ',\n' \
+                                    + sstr(self.represent[3]) + ')'
+
+    def __repr__(self):
+        return sstr(self)
+
+
+class TransferFunctionModel(object):
```

Comment:
A transfer function is a ratio of polynomials in essence, shouldn't this be a subclass that reflects that? i.e. make this a mathematical object at least based on sympy core classes.

Reply:
If it is a ratio of polynomials then we already have a way to represent that using `Mul` etc.

Reply:
> make this a mathematical object at least based on sympy core classes.

So, we can inherit `Basic`?



> If it is a ratio of polynomials then we already have a way to represent that using `Mul` etc.

How to do that?